core/Archive/ArchiveInvalidator.php

```diff
@@ -654,6 +654,10 @@ public function removeInvalidationsFromDistributedList($idSites, $pluginName = n
         $list = new ReArchiveList();
         $entries = $list->getAll();
 
+        // Make sure that idSites is an array or all before using it
+        $idSites = $idSites === 0 || $idSites === '0' || $idSites == 'all'
```

Comment:
After a quick check for how `'all'` sites are handled elsewhere this might be shorter
```suggestion
        $idSites = empty($idSites) || 'all' === $idSites
```